packages/jellyfish-api-core/__tests__/category/wallet.test.ts

```diff
@@ -17,16 +18,224 @@ describe('non masternode', () => {
+        expect(walletInfo.scanning).toBe(false)
+      })
+    })
+  })
+
+  describe('setWalletFlag', () => {
+    it('should setWalletFlag', async () => {
+      return await waitForExpect(async () => {
+        const walletInfoBefore = await client.wallet.getWalletInfo()
+        expect(walletInfoBefore.avoid_reuse).toBe(false)
+
+        const result = await client.wallet.setWalletFlag(WalletFlag.AVOID_REUSE)
+        expect(result.flag_name).toBe('avoid_reuse')
+        expect(result.flag_state).toBe(true)
+        expect(result.warnings).toBe('You need to rescan the blockchain in order to correctly mark used destinations in the past. Until this is done, some destinations may be considered unused, even if the opposite is the case.')
+
+        const walletInfoAfter = await client.wallet.getWalletInfo()
+        expect(walletInfoAfter.avoid_reuse).toBe(true)
+      })
+    })
+  })
+
+  describe('createWallet', () => {
+    it('should createWallet', async () => {
+      return await waitForExpect(async () => {
+        const wallet = await client.wallet.createWallet('alice')
+
+        expect(wallet.name).toBe('alice')
+        expect(wallet.warning).toBe('Empty string given as passphrase, wallet will not be encrypted.')
+      })
+    })
+
+    it('should createWallet with disablePrivateKeys true', async () => {
+      return await waitForExpect(async () => {
+        const wallet = await client.wallet.createWallet('bob', true)
+
+        expect(wallet.name).toBe('bob')
+        expect(wallet.warning).toBe('Empty string given as passphrase, wallet will not be encrypted.')
+      })
+    })
+
+    it('should createWallet with blank true', async () => {
+      return await waitForExpect(async () => {
+        const options = { blank: true }
+        const wallet = await client.wallet.createWallet('charlie', false, options)
+
+        expect(wallet.name).toBe('charlie')
+        expect(wallet.warning).toBe('Empty string given as passphrase, wallet will not be encrypted.')
+      })
+    })
+
+    it('should createWallet with passphrase', async () => {
+      return await waitForExpect(async () => {
+        const options = { passphrase: 'shhh...' }
+        const wallet = await client.wallet.createWallet('david', false, options)
+
+        expect(wallet.name).toBe('david')
+        expect(wallet.warning).toBe('')
+      })
+    })
+
+    it('should createWallet with avoidReuse true', async () => {
+      return await waitForExpect(async () => {
+        const options = { avoidReuse: true }
+        const wallet = await client.wallet.createWallet('eve', false, options)
+
+        expect(wallet.name).toBe('eve')
+        expect(wallet.warning).toBe('Empty string given as passphrase, wallet will not be encrypted.')
+      })
+    })
   })
 })
 
 describe('masternode', () => {
   const container = new MasterNodeRegTestContainer()
   const client = new ContainerAdapterClient(container)
+  let aliceAddress = ''
```

Comment:
should not do this, you are creating an unnecessary side effect for a test condition you specified below. The side effect being aliceAddress needs to be available when it hit that test condition.

The rule of thumb is each `it()` must be able to run on it's own.

You might want to put the aliceAddress logic into a beforeEach or create it within the test cases. 

#### When to co-locate into the test case?
When you need to isolate each sequence of event into a test scenario. Making the test more deterministic and predictable. You can then follow a linear sequence and see it performed within the block. Thinking of a flow, sequence based testing.

#### When to use beforeEach?
Especially for state based or stateful testing, you require a state to be reached before to run a test or a series of tests. Then the logic of preparing the state should be in beforeEach/All. That's where the describe blocks becomes useful.